staging/src/k8s.io/legacy-cloud-providers/gce/gce_instancegroup.go

```diff
@@ -50,6 +50,16 @@ func (g *Cloud) DeleteInstanceGroup(name string, zone string) error {
 	return mc.Observe(g.c.InstanceGroups().Delete(ctx, meta.ZonalKey(name, zone)))
 }
 
+// FilterInstanceGroupsByName lists all InstanceGroups in the project and
+// zone that match the name regexp.
+func (g *Cloud) FilterInstanceGroupsByName(name, zone string) ([]*compute.InstanceGroup, error) {
+	ctx, cancel := cloud.ContextWithCallTimeout()
+	defer cancel()
+	mc := newInstanceGroupMetricContext("filter", zone)
+	v, err := g.c.InstanceGroups().List(ctx, zone, filter.Regexp("name", name))
```

Comment:
did you check that List with regexp name works? Some of the cloud APIs don't understand the filter queries...

Reply:
fixed